src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -302,6 +301,17 @@ let add_breadcrumb_exn t breadcrumb =
         , `Int (List.length @@ Full_frontier.all_breadcrumbs t.full_frontier)
         ) ]
     "POST: ($state_hash, $n)" ;
+  let user_cmds =
+    Breadcrumb.user_commands @@ Full_frontier.best_tip t.full_frontier
+  in
+  if not (List.is_empty user_cmds) then
+    [%log' trace t.logger]
+      ~metadata:
+        [ ( "user_commands"
+          , `List
+              (List.map user_cmds
+                 ~f:(With_status.to_yojson User_command.to_yojson)) ) ]
+      "Added breadcrumb user commands" ;
```

Comment:
Can you turn this into a structured log? We aren't using the structured event ids in the filters quite yet for the test framework, but the plan is to use those instead of filtering for the message strings.